crates/api_models/src/user.rs

```diff
@@ -356,6 +356,17 @@ pub struct AuthMethodDetails {
     pub name: Option<OpenIdProvider>,
 }
 
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
+pub struct GetSsoAuthUrlRequest {
+    pub id: String,
+}
+
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
+pub struct SsoSignInRequest {
+    pub state: Secret<String>,
+    pub code: Secret<String>,
+}
+
```

Comment:
not for now. Right now we have different signin path for openidconnect signin.
We can also have type name to be openidconnect, but we can do that change later also since it wont affect backwards compatibility.